src/datatypes/interval.coffee

```diff
@@ -291,17 +292,46 @@ module.exports.Interval = class Interval
     startGreaterThanOrEqual && endEqual
 
   width: () ->
-    if @low instanceof DateTime or @high instanceof DateTime
-      throw new Error("Width of DateTime intervals is not supported")
+    if (@low? and (@low.isDateTime or @low.isDate or @low.isTime)) or
+       (@high? and (@high.isDateTime or @high.isDate or @high.isTime))
+      throw new Error("Width of Date, DateTime, and Time intervals is not supported")
 
     closed = @toClosed()
     if closed.low instanceof Uncertainty or closed.high instanceof Uncertainty
       null
+    else if closed.low.isQuantity
+      if closed.low.unit != closed.high.unit
+        throw new Error("Cannot calculate width of Quantity Interval with different units")
```

Comment:
Nothing prohibits it but the spec does say that unit conversion is optional for quantity operators.  We do support conversion for some quantity operators, but since we are in a time crunch, we opted for this route here.

Width/Size may be simple enough to support conversion for--we can give it a try if we wrap up the other changes in time.

Reply:
Created https://jira.mitre.org/browse/BONNIE-2072 @jbradl11 for tracking